haystack/document_stores/elasticsearch.py

```diff
@@ -1416,6 +1416,12 @@ def update_embeddings(
             for result_batch in get_batches_from_generator(result, batch_size):
                 document_batch = [self._convert_es_hit_to_document(hit, return_embedding=False) for hit in result_batch]
                 embeddings = retriever.embed_documents(document_batch)  # type: ignore
+                if len(embeddings[0]) != self.embedding_dim:
+                    raise ValueError(
+                        "The shape of the DocumentStore index doesn't match with the shape of the embeddings from the Transformer. Please reinitiate your DocumentStore wth embedding_dim={}.".format(
+                            len(embeddings[0])
+                        )
+                    )
                 assert len(document_batch) == len(embeddings)
```

Comment:
At the risk of going a bit off topic for the PR, since we're here I would also get rid of these `assert`s, I see in the memory docstore we raise a `DocumentstoreError` - we can maybe align all the docstores to this?

Reply:
Understood, will work on the mentioned changes. Will figure out how to write the proper test cases and add them as well. Thanks for you time !!

Reply:
I just noticed the issue resolved in memory docstore, will align my fixes to other docstores like the same.

Reply:
Since this is a negative case test, recreating it becomes tedious with various models and mismatching the embedding dim for each doc store. Please advise. @masci

Reply:
If we can't produce a fast unit test with the use of mocks, let's _not_ do it 👍